hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/FederationClientInterceptor.java

```diff
@@ -230,6 +230,9 @@ public void init(String userName) {
     this.executorService = new ThreadPoolExecutor(numMinThreads, numMaxThreads,
         keepAliveTime, TimeUnit.MILLISECONDS, workQueue, threadFactory);
 
+    // Adding this line so that unused user threads will exit and be cleaned up if idle for too long
+    this.executorService.allowCoreThreadTimeOut(true);
```

Comment:
There's a checkstyle warning here?

Reply:
Yea, seems there is some extra whitespace, removed them.